pyke/lightcurve.py

```diff
@@ -909,15 +912,42 @@ def mean_model(*theta):
             coeffs = np.asarray(theta)
             return np.dot(coeffs, cbv_array)
 
-        loss = self.loss_function(data=norm_sap_flux, mean=mean_model,
-                                  var=norm_err_sap_flux)
-        self._opt_result = loss.fit(x0=np.zeros(len(cbvs)), method=method,
-                                    options=options)
+        prior = self.prior(mean=np.zeros(len(cbvs)), var=16.)
+        likelihood = self.likelihood(data=norm_sap_flux, mean=mean_model,
+                                     var=norm_err_sap_flux)
+        x0 = likelihood.fit(x0=prior.mean, method=method, options=options).x
+        posterior = oktopus.Posterior(likelihood=likelihood, prior=prior)
+
+        self._opt_result = posterior.fit(x0=x0, method=method,
+                                         options=options)
         self._coeffs = self._opt_result.x
         flux_hat = sap_lc.flux - median_sap_flux * mean_model(self._coeffs)
-
         return LightCurve(time=sap_lc.time, flux=flux_hat.reshape(-1))
 
+    def get_cbvs_list(self, method='bayes-factor'):
+        """Returns the subsequence of subsequent CBVs that maximizes
+        Bayes' factor.
+
+        Returns
+        -------
+        cbv_list : list
+            Subsequence of subsequent CBVs that maximizes the Bayes' factor.
+        """
+
+        self.bayes_factor, cost = [], [] # bayes_factor here is actually the
+                                         # negative log of the bayes factor
+        self.correct(cbvs=[1], options={'xtol': 1e-6, 'ftol':1e-6, 'maxfev': 2000})
+        cost.append(self.opt_result.fun)
+        for n in tqdm(range(2, self._ncbvs+1)):
+            cbv_list = list(range(1, n+1))
+            self.correct(cbv_list, options={'xtol': 1e-6, 'ftol':1e-6, 'maxfev': 2000})
+            cost.append(self.opt_result.fun)
+            self.bayes_factor.append((cost[n-2] - cost[n-1]))
```

Comment:
Do you want to add a few lines of comments just to explain e.g. why `cost[n-2] - cost[n-1]` gives you the posterior ratio?  It will help future generations read the code!

Reply:
sure!